### DISCUSSION TITLE
Anyone had luck self-hosting?

### DISCUSSION THREAD

**Discussion by caudurodev** [nhost/nhost#3090] [Q&A]
I'm interested in this option for non-commercial projects. Has anyone managed to create a project and deploy simply to open source? Is there a way to adapt what is created by the CLI to self host or is this a tedious slog process of adapting everything?

There is a docker-compose.yaml in the .nhost folder - can this be modified simply to self-host?  

If not, I will consider creating a docker-compose from scratch instead with hasura, postgres, etc.

**Comment by kevin4dhd** [comment]
I've been trying to get it to work on my VPS for over a week now, no luck, there's a Docker compose example but it's from a year ago, it brings a lot of outdated stuff

The only thing I could think of is to fork the CLI, after that, since that's what works best so far, then modify the source code to make it work on a VPS.

I asked the question on discord and nhost cli is not available for self hosting, so it's up to us to modify the source code to make it work.

Maybe you already got lucky and managed to do it, I would love to know if you managed to get a Docker compose with the latest versions working for a VPS

**Reply by caudurodev** [reply]
I had a quick stab - but stopped when seeing we need a migration that creates the auth and storage tables when first spinning up postgres/hasura so did not have time to look into this. I am interested in getting this working too...
